question21/main.js

```diff
@@ -0,0 +1,159 @@
+  wrapper.appendChild(table).appendChild(getTableHeadElement(tableHeadValue));
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableRowFragment(userData));
+  sortButtonClickEvent(userData);
+}
+
+const getTableHeadElement = (tableHeadValue) => {
+  const thead = createElementWithClassName("thead", "thead");
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  for (const value of tableHeadValue) {
+    const th = createElementWithClassName("th", "th");
+    th.textContent = value;
+    value === 'ID' && th.appendChild(createSortButton());
+    fragmentThElement.appendChild(tr).appendChild(th);
+  }
+  thead.appendChild(fragmentThElement);
+  return thead;
+}
+
+const getTableRowFragment = (userData) => {
+  const tableRowfragment = document.createDocumentFragment();
+  for (let i = 0; i < userData.length; i++) {
+    const tr = createElementWithClassName("tr", "tr");
+    tableRowfragment.appendChild(tr).appendChild(getTableDataFragment(userData[i]));
+  }
+  return tableRowfragment;
+}
+
+const getTableDataFragment = (userData) => {
+  const tableDatafragment = document.createDocumentFragment();
+  Object.keys(userDataColumn).forEach((key) => {
+    const td = createElementWithClassName("td", "td");
+    td.textContent = userData[key];
+    tableDatafragment.appendChild(td);
+  });
+  return tableDatafragment;
+};
+
+const createSortButton = () => {
+  const sortButton = createElementWithClassName("button", "sortButton");
+  sortButton.dataset.status = "default";
+  sortButton.id = "js-sortbtn";
+  return sortButton;
+}
+
+const sortButtonClickEvent = (userData) => {
+  const tbody = document.querySelector('tbody');
+  const sortButton = document.getElementById('js-sortbtn');
+  sortButton.addEventListener('click', (e) => {
+    const nextStatus = getNextStatus(e.target);
+    sortButton.dataset.status = nextStatus;
+    const sortResultRows = sortFunc(e.target,userData);
+    while (tbody.firstChild) {
+      tbody.removeChild(tbody.firstChild);
+  }
+    sortResultRows.forEach((row) => {
+      tbody.appendChild(row);
+    });
+  });
+}
+
+const getNextStatus = (target) => {
+  switch (target.dataset.status) {
+    case 'default':
+      return 'asc';
+    case 'asc':
+      return 'desc';
+    case 'desc':
+      return 'default'; 
+    default:
+      return 'default';
+  }
+}
+
+const sortFunc = (target,userData) => {
+  const trElement = [...document.querySelectorAll('tbody > tr')];
+  const currentStatus = target.dataset.status;
+  const initialTrArray = [];
+  initialTrArray.push(getTableRowFragment(userData)); 
+  if (currentStatus === "default") {
+      return initialTrArray;
+  }
+  if (currentStatus === "asc"){
+      return trElement.sort((a,b) => a.children[0].textContent - b.children[0].textContent);
+  } 
+  if (currentStatus === "desc"){
+      return trElement.sort((a,b) => b.children[0].textContent - a.children[0].textContent);
+  } 
```

Comment:
IMO
In this case, the first item on the first line is the target of sort, so it should not be a problem, but it would be better to get the index dynamically.

Reply:
Thank you for your review!
That's right.I got the index of the clicked cell. Please confirm.
fixed commit: https://github.com/csakyo/homework/pull/36/commits/385bd8e957b4caa692a2a46786d89639d8d73a7a

Reply:
Oh...that's a great idea.
I got to know cellIndex!